### ISSUE TITLE
感情分析結果のDB保存と可視化（時系列対応）

### ISSUE THREAD

**Issue by rindguitar** [rindguitar/game-demand-forecast#10]
## 概要

Issue #6で学習した感情分析モデルの結果をDBに保存し、可視化する基盤を構築する。

**重要**: 収集済みデータに`timestamp_created`が含まれているため、時系列分析にも対応する。

---

### 現状
- ✅ 感情分析モデル完成（Test Acc 85.70%）
- ✅ 4984件のレビューデータあり（`data/train/reviews_5000.csv`）
- ✅ `timestamp_created`（投稿日時）を含むデータ収集済み
- ❌ 分析結果がDB保存されていない
- ❌ 可視化機能なし

### 目的
1. 感情分析結果をDBに永続化（再分析不要）
2. 静的な集計グラフ（ゲーム比較）
3. **時系列グラフ（トレンド分析）** ← NEW!
4. Prophet/LSTM実装の準備（Phase 7-8）

---

#### タスク1-1: DBスキーマ設計
- [ ] `sentiment_results`テーブル作成
  ```sql
  CREATE TABLE sentiment_results (
      id INTEGER PRIMARY KEY AUTOINCREMENT,
      app_id INTEGER NOT NULL,
      game_name TEXT,
      review_text TEXT NOT NULL,
      timestamp_created INTEGER,  -- Unixタイムスタンプ
      sentiment TEXT NOT NULL,    -- 'POSITIVE' or 'NEGATIVE'
      sentiment_score REAL,       -- 0.0-1.0（確信度）
      analyzed_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
      model_version TEXT DEFAULT 'distilbert-phase2'
  );
  
  CREATE INDEX idx_app_id ON sentiment_results(app_id);
  CREATE INDEX idx_timestamp ON sentiment_results(timestamp_created);
  CREATE INDEX idx_sentiment ON sentiment_results(sentiment);
  ```

#### タスク1-2: 感情分析 → DB保存スクリプト
- [ ] `scripts/analyze_and_save_to_db.py`作成
  - CSVからレビュー読み込み（`timestamp_created`含む）
  - Issue #6のモデルで感情分析
  - 結果をDBに保存
  - 進捗表示（何件中何件完了）

#### タスク1-3: 動作確認
- [ ] 4984件のレビューを分析してDB保存
- [ ] DBに正しく保存されたか確認

---

#### タスク2-1: 基本クエリ関数実装
- [ ] `src/nlp/sentiment_db.py`作成
  - `get_sentiment_by_game(app_id)`: ゲーム別の感情分析結果
  - `get_all_sentiments()`: 全ゲームの感情分析結果
  - `get_sentiment_stats(app_id)`: ゲームの感情統計（Positive/Negative比率）

#### タスク2-2: 時系列クエリ関数実装
- [ ] `get_sentiment_timeseries(app_id, interval)`: 時系列データ取得
  - `interval`: 'day', 'week', 'month'
  - 戻り値: DataFrame（date, positive_rate, review_count）
  - SQLiteの`DATE()`関数でUnixタイムスタンプを変換

#### タスク2-3: 動作確認
- [ ] 各関数の単体テスト
- [ ] 7ゲーム全てのデータを正しく取得できるか確認

---

#### タスク3-1: 静的グラフ（集計）
- [ ] `src/visualization/sentiment_plots.py`作成
  - `plot_sentiment_distribution(app_id)`: ゲーム別の感情分布（円グラフ or 棒グラフ）
  - `plot_sentiment_by_game()`: 全ゲーム比較（棒グラフ）

**例**:
```
Baldur's Gate 3:  [████████████░░] 92% Positive
Stardew Valley:   [███████████░░░] 89% Positive
Cyberpunk 2077:   [████████░░░░░░] 72% Positive
...
```

#### タスク3-2: 時系列グラフ
- [ ] `plot_sentiment_timeseries(app_id, interval)`: 感情スコア時系列グラフ
  - 日/週/月ごとのPositive割合をプロット
  - トレンド分析（上昇/下降）
  - アップデート前後の比較

**例: Cyberpunk 2077のアップデート前後**
```
Positive Rate (%)
100 ┤
 90 ┤              ┌─ アップデート後
 80 ┤            ╱
 70 ┤         ╱
 60 ┤──────┘
 50 ┤  アップデート前
    └─────────────────────────
    2023-06  2023-08  2023-10
```

#### タスク3-3: 保存・エクスポート機能
- [ ] グラフをPNG保存
- [ ] 統計レポートをMarkdown出力

---

### 定量目標
- [x] 4984件のレビューを分析してDB保存
- [x] 7ゲームの感情分布を可視化
- [x] 7ゲームの時系列グラフを生成

### 定性目標
- [x] 感情分析結果の再利用が可能（再分析不要）
- [x] ゲーム間の感情比較が可能
- [x] 感情の時間変化を把握できる
- [x] Prophet/LSTM実装の準備完了

---

### データソース
- `data/train/reviews_5000.csv`（4984件、NaN除去済み）
- カラム: `review_text`, `game_id`, `game_name`, `timestamp_created`, `label`

### モデル
- Issue #6で学習したDistilBERTモデル
- パス: `models/sentiment_model_phase2/model.pth`
- Test Accuracy: 85.70%

### データベース
- SQLite: `data/game_demand.db`
- 新規テーブル: `sentiment_results`

### 可視化ライブラリ
- Matplotlib（基本グラフ）
- Seaborn（統計グラフ）
- Pandas（データ処理）

---

## 実装規模見積もり

| Phase | 実装量 | 予想時間 |
|-------|--------|----------|
| Phase 1 | 中規模 | 2-3時間 |
| Phase 2 | 小規模 | 1時間 |
| Phase 3 | 中規模 | 2時間 |
| **合計** | - | **5-6時間** |

---

### コード
- `scripts/analyze_and_save_to_db.py`: 感情分析 → DB保存スクリプト
- `src/nlp/sentiment_db.py`: DBクエリ関数
- `src/visualization/sentiment_plots.py`: 可視化関数

### データ
- `data/game_demand.db`: 感情分析結果を含むSQLiteデータベース

### 可視化
- `outputs/sentiment_distribution_*.png`: ゲーム別感情分布グラフ
- `outputs/sentiment_comparison.png`: 全ゲーム比較グラフ
- `outputs/sentiment_timeseries_*.png`: 時系列グラフ（7ゲーム）

### ドキュメント
- `outputs/sentiment_report.md`: 統計レポート

---

## 関連リンク

- **Issue #6**: PyTorch感情分析モデルをゼロから学習（完了）
- **Wiki**: [過学習（Overfitting）](https://github.com/rindguitar/game-demand-forecast/wiki/Overfitting)
- **Wiki**: [Train/Val/Test Split](https://github.com/rindguitar/game-demand-forecast/wiki/Train-Val-Test-Split)
- **CLAUDE.md**: データフロー（NLP処理 → 時系列データ準備 → 予測）

---

## 次のステップ（このIssue完了後）

1. **Phase 4**: トピック抽出（BERTopic）
2. **Phase 7**: Prophet実装（時系列予測）
3. **Phase 8-9**: LSTM実装（時系列予測）

このIssueは**Phase 7-8の準備**として重要です。
